File: bidrag-commons-test/src/main/kotlin/no/nav/bidrag/commons/testdatagenerering/person/IdentType.java
package no.nav.bidrag.commons.testdatagenerering.person;

import java.time.LocalDate;

import static no.nav.bidrag.commons.testdatagenerering.RandomTestData.random;
import static no.nav.bidrag.commons.testdatagenerering.person.Kjonn.KVINNE;
import static no.nav.bidrag.commons.testdatagenerering.person.Kjonn.MANN;


public interface IdentType {
    String generer(LocalDate fodtDato, Kjonn kjonn);

    default String generer() {
        return generer(
                random().dateBetween(LocalDate.of(1900, 1, 1), LocalDate.now()),
                random().oneOf(KVINNE, MANN));
    }
}

File: bidrag-generer-testdata/src/main/kotlin/no.nav.bidrag.generer.testdata/person/IdentType.kt
package no.nav.bidrag.generer.testdata.person

import no.nav.bidrag.generer.testdata.RandomTestData
import java.time.LocalDate

@Suppress("unused")
interface IdentType {
    fun generer(
        fodtDato: LocalDate,
        kjønn: Kjønn?,
    ): String?

    fun generer(): String? =
        generer(
            RandomTestData.Companion.random().dateBetween(LocalDate.of(1900, 1, 1), LocalDate.now()),
            Kjønn.entries.random()
        )
}
